test/elements.js

```diff
@@ -104,6 +104,53 @@ test('adjacent text nodes', function (t) {
   t.end()
 })
 
+test('space in only-child text nodes', function (t) {
+  t.plan(1)
+  var result = bel`
+    <span>
+      surrounding
+      newlines
+    </span>
+  `
+  t.equal(result.outerHTML, '<span>surrounding\n      newlines</span>', 'should remove extra space')
+  t.end()
+})
+
+test('space between text and non-text nodes', function (t) {
+  t.plan(1)
+  var result = bel`
+    <div>
+      <dfn>whitespace</dfn>
+      is empty
+    </div>
+  `
+  t.equal(result.outerHTML, '<div><dfn>whitespace</dfn> is empty</div>', 'should have correct output')
+  t.end()
+})
+
+test('space between non-text nodes', function (t) {
+  t.plan(1)
+  var result = bel`
+    <div>
+      <dfn>whitespace</dfn>
+      <em>is beautiful</em>
+    </div>
+  `
+  t.equal(result.outerHTML, '<div><dfn>whitespace</dfn> <em>is beautiful</em></div>', 'should have correct output')
```

Comment:
perhaps if we apply the heuristic that if "inline-text nodes" follow each other (or an actual text node), we add a space. E.g.
```html
<strong>hi</strong>
<em>hello</em>
<span>world</span>
planet
<div>bar</div>
```
```html
<strong>hi</strong> <em>hello</em> <span>world</span> planet<div>bar</div>
```

I don't think we can account for all edge cases (html is huuge); but as long as common patterns are supported then we might get away with it (: